packages/auth/AuthStore.ts

```diff
@@ -0,0 +1,126 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2022 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import createAuth0Client, {
+  Auth0Client,
+  Auth0ClientOptions,
+  GetTokenSilentlyOptions,
+  User,
+} from "@auth0/auth0-spa-js";
+import { makeAutoObservable, runInAction } from "mobx";
+import qs from "qs";
+
+interface AuthStoreProps {
+  authSettings: Auth0ClientOptions | undefined;
+}
+export class AuthStore {
+  readonly authSettings: Auth0ClientOptions | undefined;
+
+  private authClient: Auth0Client | undefined;
+
+  emailVerified?: boolean;
+
+  isAuthorized: boolean;
+
+  isLoading: boolean;
+
+  user?: User;
+
+  constructor({ authSettings }: AuthStoreProps) {
+    makeAutoObservable(this);
+
+    this.authSettings = authSettings;
+    this.authClient = undefined;
+    this.isAuthorized = false;
+    this.isLoading = true;
+  }
+
+  private get auth0Client(): Promise<Auth0Client> {
+    return createAuth0Client(this.authSettings as Auth0ClientOptions);
```

Comment:
assuming there is a legitimate use case for `this.authSettings` being undefined we probably want to set an error state or return undefined or something when that happens, rather than making a type assertion

Reply:
Great call! So adding a new error member to this class?